src/defs.h

```diff
@@ -485,26 +485,30 @@ struct sockaddr_store_t {
  * @brief Socket related info
  */
 
-typedef struct fds_data {
+struct fds_data {
     struct sockaddr_store_t server_addr; /**< server address information */
-    socklen_t server_addr_len;      /**< server address length */
-    int is_multicast;               /**< if this socket is multicast */
-    int sock_type;                  /**< SOCK_STREAM (tcp), SOCK_DGRAM (udp), SOCK_RAW (ip) */
-    int next_fd;
-    int active_fd_count; /**< number of active connections (by default 1-for UDP; 0-for TCP) */
-    int *
-    active_fd_list; /**< list of fd related active connections (UDP has the same fd by default) */
-    struct sockaddr_store_t *memberships_addr; /**< more servers on the same socket information */
+    socklen_t server_addr_len = 0;  /**< server address length */
+    int is_multicast = 0;           /**< if this socket is multicast */
+    int sock_type = 0;              /**< SOCK_STREAM (tcp), SOCK_DGRAM (udp), SOCK_RAW (ip) */
+    int next_fd = 0;
+    int active_fd_count = 0;        /**< number of active connections (by default 1-for UDP; 0-for TCP) */
+    int *active_fd_list = nullptr;  /**< list of fd related active connections (UDP has the same fd by default) */
+    struct sockaddr_store_t *memberships_addr = nullptr; /**< more servers on the same socket information */
     IPAddress mc_source_ip_addr;    /**< message source ip for multicast packet filtering */
-    int memberships_size;
+    int memberships_size = 0;
     struct SocketRecvData recv;
 #ifdef USING_VMA_EXTRA_API
-    Message *p_msg;
+    Message *p_msg = nullptr;
 #endif
 #if defined(DEFINED_TLS)
-    void *tls_handle;
+    void *tls_handle = nullptr;
 #endif /* DEFINED_TLS */
-} fds_data;
+
+    fds_data()
+    {
+        memset(&server_addr, 0, sizeof(server_addr));
+    }
+};
```

Comment:
please consider just `memset(this, 0, sizeof(*this));` instead of doing initialization explicitly for every member as far as all values are 0.

Reply:
This may result in a weird behavior of mc_source_ip_addr member (of type IPAddress). This class will probably be extended with std::string member in PR #178.